### ISSUE TITLE
Android adyen Wechat Pay Component callBack

### ISSUE THREAD

**Issue by xiaoxiaohua333** [Adyen/adyen-android#204]
**Describe the bug**

when handle wechat action, the Wechat PayReq options callbackClassName is wrong!
Don't use activity.getLocalClassName(), you need use activity.getClass().getName()

 ```java
   
    @Override
    protected void handleActionInternal(@NonNull Activity activity, @NonNull Action action) throws ComponentException {
        Logger.d(TAG, "handleActionInternal: activity - " + activity.getLocalClassName());
        //noinspection unchecked
        final SdkAction<WeChatPaySdkData> weChatAction = (SdkAction<WeChatPaySdkData>) action;
        if (weChatAction.getSdkData() != null) {
            final boolean weChatInitiated = initiateWeChatPayRedirect(weChatAction.getSdkData(), activity.getLocalClassName());
            if (!weChatInitiated) {
                throw new ComponentException("Failed to initialize WeChat app.");
            }
        } else {
            throw new ComponentException("WeChatPay Data not found.");
        }
    }

    private boolean initiateWeChatPayRedirect(@NonNull WeChatPaySdkData weChatPaySdkData, @NonNull String callbackActivityName) {
        Logger.d(TAG, "initiateWeChatPayRedirect");
        mApi.registerApp(weChatPaySdkData.getAppid());
        return mApi.sendReq(WeChatPayUtils.generatePayRequest(weChatPaySdkData, callbackActivityName));
    }


    /**
     * Returns class name for this activity with the package prefix removed.
     * This is the default name used to read and write settings.
     *
     * @return The local class name.
     */
    @NonNull
    public String getLocalClassName() {
        final String pkg = getPackageName();
        final String cls = mComponent.getClassName();
        int packageLen = pkg.length();
        if (!cls.startsWith(pkg) || cls.length() <= packageLen
                || cls.charAt(packageLen) != '.') {
            return cls;
        }
        return cls.substring(packageLen+1);
    }
```

**Smartphone (please complete the following information):**
 - Device: [e.g. One Plus 6T]
 - OS: [e.g.android 10]

**Additional context**
 implementation "com.adyen.checkout:wechatpay-base:3.6.6"

**Comment by caiofaustino** [comment]
hi @xiaoxiaohua333,

Thanks for pointing this out, but on our tests and other merchants using the SDK we didn't have any problems with this.
Can you point to the WeChat SDK specification saying it should be a full class name with package instead of simple name?
I just want to confirm this with a more reliable source before making the change on our end.

**Comment by xiaoxiaohua333** [comment]
hi @caiofaustino ,

Can you try to update Wechat sdk to  last version (com.tencent.mm.opensdk:wechat-sdk-android-without-mta:+ , last version is 6.6.4), I found adyen use ext.wechat_pay_version = "5.1.6".

Now we need use Wechat sdk last version, I only Override getLocalClassName() method to get Wechat callback data

``` java

    @NonNull
    @Override
    public String getLocalClassName() {
        return super.getClass().getName();
    }

```

**Comment by caiofaustino** [comment]
Yeah we will work on that. Thanks

**Comment by caiofaustino** [comment]
Hi @xiaoxiaohua333,

This fix has been made on version `4.0.0-alpha01` if you want to test it. On that version we also update to AndroidX libraries. A stable release of version 4.0 might still take a little while, but let us know if this helps.

Cheers
